### DISCUSSION TITLE
Implementing + and - labels

### DISCUSSION THREAD

**Discussion by MartsINY1** [cc65/cc65#1864] [Ideas]
Those labels, if implemented, would be really useful for those use to them and they wouldn't conflict with anything

**Comment by polluks** [comment]
What about [unnamed labels](https://cc65.github.io/doc/ca65.html#ss6.6)?

**Reply by MartsINY1** [reply]
the problem is if adding a : somewhere, need to see all branchings close, while if using +, ++, +++, you only need to add a new one that is unused.
Also, implementing this option could coexist with the rest, so it offers another option for people who prefer using it

**Reply by MartsINY1** [reply]
> What about [unnamed labels](https://cc65.github.io/doc/ca65.html#ss6.6)?

@polluks so is there a posssibility to implement it? would be really useful, as explained by Fiskbit

**Reply by polluks** [reply]
Maybe... I don't implement features.

**Reply by MartsINY1** [reply]
@polluks I'm no expert, do you know how to get attention of the features implementers?

**Comment by Fiskbit** [comment]
There are multiple different styles of anonymous labels. The + and - version is a pretty common one, at least across assemblers used by NES developers. The : style used by ca65 is the most fragile style I've encountered because the labels are so interconnected; adding or removing :'s as you modify the code can cause nearby branches to go to the wrong place, so they're very easy to break without noticing. It's also difficult when multiple are in use to tell what branches are going where. In comparison, with + and -, you can choose a label length (e.g. +++) that is not used nearby, and the ability to have different label names makes it easier to tell at a glance what is being referenced. The labels also have a nice benefit in making it clear which direction they're used from, with - labels often being a good indicator of a loop. I find myself using anonymous labels much less in ca65 than I normally would because of the : style's many drawbacks.

A third style I've used and which is easily my favorite is numbered anonymous labels in GAS. Here, the label is just a number, and is referenced by the number followed by f for forward or b for backward (e.g. 1f, 3b), which references the first label in that direction of that number. I've been hoping to be able to use these in a 6502 assembler, though the + and - version is a decent substitute.

I think supporting + and - would both reduce the risk associated with using anonymous labels and also make it much easier for people to switch their projects to ca65 from assemblers such as asm6, as converting labels in a large project can be the bulk of the work and easy to mess up.

**Comment by mrdudz** [comment]
I'd be willing to merge a PR that implements this (and respects codestyle etc - see contributing.md). I don't have time to work on this however.

**Reply by MartsINY1** [reply]
Not something I would know how to make :(
Hope someone who cab will or it will be added as a feature to be implemented, would lobe to try switch to this program, people say its better

**Reply by MartsINY1** [reply]
where should I write a request if I would like it to be in a feature to be implemented list?

**Reply by cherrr** [reply]
ho yes!!! are they finally implementing this function? I always felt like it was something simple missing from this assembler.

**Comment by mrdudz** [comment]
Not sure what the question is - we dont have a dedicated list for that kind of thing :)

**Reply by MartsINY1** [reply]
thanks a lot for your answer. So if I want to suggest something to be sure whole team could decide if they want to implement it or not, what would be the best way?

**Reply by rofl0r** [reply]
supposedly the whole "team" was reading this conversation but nobody feels like doing the work for you... your best option is to implement it yourself and file a PR

**Reply by MartsINY1** [reply]
sorry my bad. I must misunderstand how github works, I expected project to receive suggestions, since not eveybody know how to code. I understand github is more of a do your own, and I can I program assembly only, just ignore my suggestion
